# Tuberías en paralelo
## Comprobación de diseño
### **Ejemplo No.1**
> En la red matriz del sistema de acueducto del municipio de Santa Marta, Colombia, existen dos tuberias que unen la planta de tratamiento de Mamatoco y el tanque de las Tres Cruces. Las dos tuberias tienen una longitud de 627 m y un coeficiente global de perdidas menores de 10.6. Una de ellas tiene un diametro de 8 pulg en PVC ($k_s = $0.0015 mm) y la otra tiene un diametro de 12 pulg y esta fabricada en asbesto-cemento ($k_s =$0.03 mm). La diferencia de cabeza entre los nodos de aguas arriba y aguas abajo es de 26.4 m. El agua se encuentra a 20 $^o$C. Calcular el caudal total.

> - $\rho =$ 998.2 kg/m$^3$ 
> - $\mu =$ 1.005x10$^{-3}$ Pa.s 

> *Solucion:*

> Siguiendo el procedimiento descrito, se tiene:

> La imformacion de entrada al programa es:

In [1]:
import json
with open('designTest_js55.json', 'r') as f:
    data = json.load(f)
data

{'PT': 1,
 'US': 'IS',
 'IM': 'nr',
 'rho': 998.2,
 'mu': 0.001005,
 'nu': '',
 'E1': {'z': 26.4, 'p': 0, 'v': 0},
 'E2': {'z': 0, 'p': 0, 'v': 0},
 'Q': '',
 'Pu': {'P': '', 'h': '', 'ef': 1},
 'P1': {'Q': '',
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.2032,
  'L': 627,
  'ks': 1.5e-06,
  'K': [10.6]},
 'P2': {'Q': '',
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.3048,
  'L': 627,
  'ks': 3e-05,
  'K': [10.6]}}

In [2]:
%run main.py designTest_js55.json


Proving the design of a parallel pipe system

 
Results of iteration for pipe 1
 
          f         V         hf
1  0.000100  6.890769   0.746759
2  0.014637  3.047730  21.381663
3  0.012804  3.215086  20.815399
4  0.012684  3.227066  20.773704
5  0.012676  3.227868  20.770909
6  0.012675  3.227921  20.770723
7  0.012675  3.227924  20.770711
 
Results of iteration for pipe 2
 
          f         V         hf
1  0.000100  6.923486   0.502578
2  0.014444  3.584567  19.458070
3  0.013327  3.691293  19.038540
4  0.013294  3.694530  19.025625
5  0.013293  3.694625  19.025247
6  0.013293  3.694627  19.025235
 
Summary of results
 
           hf         V         Q        he         f            Re
P1  20.770711  3.227924  0.104679  5.629289  0.012675  6.514762e+05
P2  19.025235  3.694627  0.269582  7.374765  0.013293  1.118503e+06
Qt =   0.3743 m³/s


### **Ejemplo No.2**
> Calcule el caudal total que fluye por el sistema en paralelo mostrado en la figura. La presion en el nodo de entrada es de 530 kPa y en el nodo de salida es de 130 kPa, ambas manometricas. Las tuberias son de acero comercial ($k_s$ = 0.046 mm)

> $\rho$ = 999.1 kg/m$^3$

> $\mu$ = 1.14 X 10$^{-3}$ Pa.s
 
![alternatvie text](./figs/fjs513.jpeg)

> *Solucion:*

> Siguiendo el procedimiento descrito, se tiene:

> La imformacion de entrada al programa es:

In [3]:
import json
with open('designTest_js513.json', 'r') as f:
    data = json.load(f)
data

{'PT': 1,
 'US': 'IS',
 'IM': 'nr',
 'rho': 999.1,
 'mu': 0.00114,
 'nu': '',
 'E1': {'z': 0, 'p': 54.0265, 'v': 0},
 'E2': {'z': 0, 'p': 13.2518, 'v': 0},
 'Q': '',
 'Pu': {'P': '', 'h': '', 'ef': 1},
 'P1': {'Q': '',
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.1524,
  'L': 278,
  'ks': 4.6e-05,
  'K': [7.4]},
 'P2': {'Q': '',
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.2031,
  'L': 230,
  'ks': 4.6e-05,
  'K': [6.1]},
 'P3': {'Q': '',
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.1524,
  'L': 278,
  'ks': 4.6e-05,
  'K': [7.4]}}

In [4]:
%run main.py designTest_js513.json


Proving the design of a parallel pipe system

 
Results of iteration for pipe 1
 
          f          V         hf
1  0.000100  10.271672   0.980941
2  0.016825   4.582825  32.853347
3  0.016077   4.667192  32.559010
4  0.016059   4.669248  32.551769
5  0.016059   4.669297  32.551596
6  0.016059   4.669299  32.551592
 
Results of iteration for pipe 2
 
          f          V         hf
1  0.000100  11.347120   0.743174
2  0.015630   5.797719  30.324007
3  0.014938   5.895516  29.968464
4  0.014926   5.897351  29.961735
5  0.014925   5.897385  29.961611
6  0.014925   5.897385  29.961609
 
Results of iteration for pipe 3
 
          f          V         hf
1  0.000100  10.271672   0.980941
2  0.016825   4.582825  32.853347
3  0.016077   4.667192  32.559010
4  0.016059   4.669248  32.551769
5  0.016059   4.669297  32.551596
6  0.016059   4.669299  32.551592
 
Summary of results
 
           hf         V         Q         he         f            Re
P1  32.551592  4.669299  0.085175   8.2

## Calculo de la potencia requerida
### **Ejemplo No.1**
> Con el fin de disminuir la vulnerabilidad del cruce subacuatico de un oleoducto al atravesar un rio, se decide colocar dos tuberias en paralelo a la existente. Las tres tuberias estan elaboradas en acero ($k_s = $ 0.046 mm). La que existia originalmente tiene una longitud de 278 m, un diametro de 18 pulg y un coeficiente global de perdidas menores de 7.7. Las dos tuberias nuevas tienen una longitud 312 m, un diametro de 12 pulg y un coeficiente global de perdidas menores de 9.4. La presion en el nodo de aguas arriba es de 875 kPa y el caudal total es de 460 l/s. Calcular el caudal que transporta cada tuberia y la perdida de cabeza. Las caracteristicas fisicas del crudo transportado por las tuberias es:

> - $\rho$ = 860 kg/m$^3$

> - $\mu$ = 7.19 X 10$^{-3}$ Pa.s

> *Solucion:*

> Siguiendo el procedimiento descrito, se tiene:

> La imformacion de entrada al programa es:

In [5]:
import json
with open('systemPower_js56.json', 'r') as f:
    data = json.load(f)
data

{'PT': 2,
 'US': 'IS',
 'IM': 'nr',
 'rho': 860.0,
 'mu': 0.00719,
 'nu': '',
 'E1': {'z': 0, 'p': 103.7148, 'v': 0},
 'E2': {'z': 0, 'p': '', 'v': 0},
 'Q': 0.46,
 'Pu': {'P': '', 'h': '', 'ef': 1},
 'P1': {'Q': '',
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.4572,
  'L': 278,
  'ks': 4.6e-05,
  'K': [7.7]},
 'P2': {'Q': '',
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.3048,
  'L': 312,
  'ks': 4.6e-05,
  'K': [9.4]},
 'P3': {'Q': '',
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.3048,
  'L': 312,
  'ks': 4.6e-05,
  'K': [9.4]}}

In [6]:
%run main.py systemPower_js56.json


Estimating the power in parallel pipes

Results of iteration 1

          hf         V         Q        he         f            Re
P1  1.615313  1.662772  0.272983  1.085068  0.018852  90930.266584
P2  1.896857  1.295046  0.094494  0.803524  0.021678  47213.872301
P3  1.896857  1.295046  0.094494  0.803524  0.021678  47213.872301

Results of iteration 2

          hf         V         Q        he         f            Re
P1  1.602862  1.655678  0.271818  1.075830  0.018867  90542.328624
P2  1.882123  1.289428  0.094084  0.796569  0.021698  47009.076066
P3  1.882123  1.289428  0.094084  0.796569  0.021698  47009.076066

Results of iteration 3

          hf         V         Q        he         f            Re
P1  1.602947  1.655727  0.271826  1.075893  0.018867  90544.981148
P2  1.882223  1.289467  0.094087  0.796616  0.021698  47010.476318
P3  1.882223  1.289467  0.094087  0.796616  0.021698  47010.476318

Pressure head in node 2 =   101.04 (m)
Pressure in node 2 = 852399.98 (Pa)


### **Ejemplo No.2**
> En la red matriz del sistema de distribucion de agua potable del acueducto de Pereira, Colombia, se tiene el sistema en paralelo que se muestra en la figura. El caudal total que debe pasar por este es de 254.3 l/s y la presion en el nodo inicial es de 343 kPa. El material de ambas tuberias es asbesto-cemento. Determinar la presion en el nodo final y los caudales transportados por cada tuberia. 

> - $\rho$ = 999.1 kg/m$^3$
> - $\nu$ = 1.141 X 10$^{-6}$ m$^2$/s

![alternatvie text](./figs/fjs516.jpeg)

> *Solucion:*

> Siguiendo el procedimiento descrito, se tiene:

> La imformacion de entrada al programa es:

In [1]:
import json
with open('systemPower_js516.json', 'r') as f:
    data = json.load(f)
data

{'PT': 2,
 'US': 'IS',
 'IM': 'nr',
 'rho': 999.1,
 'mu': '',
 'nu': 1.141e-06,
 'E1': {'z': 0, 'p': 34.9643, 'v': 0},
 'E2': {'z': 0, 'p': '', 'v': 0},
 'Q': 0.2543,
 'Pu': {'P': '', 'h': '', 'ef': 1},
 'P1': {'Q': '',
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.254,
  'L': 263,
  'ks': 3e-05,
  'K': [7.6]},
 'P2': {'Q': '',
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.3048,
  'L': 277,
  'ks': 3e-05,
  'K': [7.4]}}

In [2]:
%run main.py systemPower_js516.json


Estimating the power in parallel pipes

Results of iteration 1

          hf         V         Q        he         f             Re
P1  3.065256  1.978145  0.100234  1.515761  0.014843  440358.321342
P2  2.912786  2.103109  0.153455  1.668231  0.014217  561812.008688

Results of iteration 2

          hf         V         Q        he         f             Re
P1  3.079156  1.982907  0.100475  1.523069  0.014839  441418.466678
P2  2.925980  2.108154  0.153823  1.676245  0.014213  563159.932053

Results of iteration 3

          hf         V         Q        he         f             Re
P1  3.079185  1.982917  0.100476  1.523084  0.014839  441420.655707
P2  2.926007  2.108165  0.153824  1.676262  0.014213  563162.715290

Pressure head in node 2 =    30.36 (m)
Pressure in node 2 = 297583.46 (Pa)


## Diseno de tuberias
### **Ejemplo No.1**
> En la planta de tratamiento de aguas residuales del municipio de Ubate, la tuberia de descarga al rio Suta tiene un diametro de 12 pulg, esta elaborada en hierro galvanizado ($k_s = $0.15 mm), tiene una longitud de 150 m y un coeficiente global de perdidas menores de 3.3. La cabeza en el nodo de entrada es de 2.7 m y la del nodo de salida es 0.5 m. Bajo estas condiciones el caudal maximo que puede fluir por la tuberia es de 138.5 l/s. Por razones de crecimiento de la industria lechera del municipio, el caudal total que debe ser tratado aumenta a 224.2 l/s. Determinar el diametro para una tuberia paralela a la primera si el material disponible es PVC, y la nueva presion en el nodo de salida. La longitud y el coeficiente global de perdidas menores de la nueva tuberia es igual al de la tuberia existente. La temperature media del agua es de 14 $^o$C.  

> - $\rho$ = 999.1 kg/m$^3$

> - $\nu$ = 1.17 X 10$^{-6}$ m$^2$/s

> *Solucion:*

> Siguiendo el procedimiento descrito, se tiene:

> La imformacion de entrada al programa es:

In [3]:
import json
with open('pipeDesign_js57.json', 'r') as f:
    data = json.load(f)
data

{'PT': 3,
 'US': 'IS',
 'IM': 'nr',
 'rho': 999.1,
 'mu': '',
 'nu': 1.17e-06,
 'E1': {'z': 0, 'p': 2.7, 'v': 0},
 'E2': {'z': 0, 'p': 0.5, 'v': 0},
 'Q': 0.2242,
 'Pu': {'P': '', 'h': '', 'ef': 1},
 'P1': {'Q': 0.1385,
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.3048,
  'L': 150,
  'ks': 0.00015,
  'K': [3.3]},
 'P2': {'Q': 0.0857,
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': '',
  'L': 150,
  'ks': 1.5e-06,
  'K': [3.3]}}

In [4]:
%run main.py pipeDesign_js57.json


Design of a parallel pipe system

Design of pipe 2
 
        D"        hf         V         Q        he         f
Iter 1   4  1.955812  1.204910  0.009769  0.244188  0.017903
Iter 2   6  1.814047  1.514816  0.027632  0.385953  0.015759
Iter 3   8  1.681489  1.755786  0.056939  0.518511  0.014497
Iter 4  10  1.560696  1.949603  0.098788  0.639304  0.013642

Results of iteration 1

          hf         V         Q        he         f
P1  1.258134  1.681697  0.122706  0.475675  0.017736
P2  1.238137  1.716681  0.086986  0.495672  0.013958

Results of iteration 2

          hf         V         Q        he         f
P1  1.424554  1.792537  0.130794  0.540444  0.017675
P2  1.398365  1.835453  0.093004  0.566633  0.013790

Results of iteration 3

          hf         V         Q        he         f
P1  1.429314  1.795610  0.131018  0.542299  0.017674
P2  1.402942  1.838752  0.093171  0.568671  0.013786

Results of iteration 4

          hf         V         Q        he         f
P1  1.42944

### **Ejemplo No.2**
> En el subsistema de distribucion de agua potable del acueducto de Pereira, Colombia, que parte del tanque Matecaña se tiene una tuberia con las caracteristicas mostradas en la figura. El caudal maximo que puede fluir por esta tuberia es 239.4 l/s. La presion en el nodo de entrada equivale a 25.3 m de agua y la del nodo final es de 17.6 m de agua. Si se quiere que el caudal aumente a 390 l/s, ¿cual debera ser el diametro de la nueva tuberia si su longitud y coeficiente global de perdidas menores son iguales a la de la tuberia original y el material es PVC? ¿Cuales son los caudales finales en cada una de las tuberias? ¿Cual es la cabeza final en el nodo 2?  

> - $\rho$ = 999.1 kg/m$^3$

> - $\nu$ = 1.141 X 10$^{-6}$ m$^2$/s


![alternatvie text](./figs/fjs519.jpeg)

> *Solucion:*

> Siguiendo el procedimiento descrito, se tiene:

> La imformacion de entrada al programa es:

In [1]:
import json
with open('pipeDesign_js519.json', 'r') as f:
    data = json.load(f)
data

{'PT': 3,
 'US': 'IS',
 'IM': 'nr',
 'rho': 999.1,
 'mu': '',
 'nu': 1.141e-06,
 'E1': {'z': 1630, 'p': 25.3, 'v': 0},
 'E2': {'z': 1573, 'p': 17.6, 'v': 0},
 'Q': 0.39,
 'Pu': {'P': '', 'h': '', 'ef': 1},
 'P1': {'Q': 0.2394,
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.3048,
  'L': 187,
  'ks': 3e-05,
  'K': [5.7]},
 'P2': {'Q': 0.1506,
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': '',
  'L': 187,
  'ks': 1.5e-06,
  'K': [5.7]}}

In [2]:
%run main.py pipeDesign_js519.json


Design of a parallel pipe system

Design of pipe 2
 
       D"         hf         V         Q         he         f
Iter 1  4  52.279542  6.538542  0.053010  12.420458  0.013035
Iter 2  6  46.376952  7.941653  0.144867  18.323048  0.011758
Iter 3  8  41.378960  8.959547  0.290552  23.321040  0.010990

Results of iteration 1

          hf         V         Q        he         f
P1  5.452749  3.598654  0.262579  3.762323  0.013465
P2  6.236184  3.202132  0.103843  2.978889  0.012966

Results of iteration 2

          hf         V         Q        he         f
P1  6.119527  3.822335  0.278900  4.244567  0.013395
P2  6.989505  3.408180  0.110525  3.374589  0.012829

Results of iteration 3

          hf         V         Q        he         f
P1  6.136267  3.827793  0.279298  4.256697  0.013393
P2  7.008401  3.413213  0.110688  3.384564  0.012825

Results of iteration 4

          hf         V         Q        he         f
P1  6.136666  3.827923  0.279308  4.256986  0.013393
P2  7.008850  3